src/systems/perception.rs

```diff
@@ -0,0 +1,70 @@
+use amethyst::{
+    core::{nalgebra::Vector4, transform::GlobalTransform},
+    ecs::{
+        Entities, Join, Read, ReadExpect, ReadStorage, System, Write, WriteExpect, WriteStorage,
+    },
+    renderer::DebugLines,
+};
+
+use crate::components::perception::Perception;
+use crate::resources::{debug::DebugConfig, spatial_grid::SpatialGrid};
+
+pub struct EntityDetectionSystem;
+
+impl<'s> System<'s> for EntityDetectionSystem {
+    type SystemData = (
+        Entities<'s>,
+        WriteStorage<'s, Perception>,
+        ReadExpect<'s, SpatialGrid>,
+        ReadStorage<'s, GlobalTransform>,
+        Write<'s, DebugLines>,
```

Comment:
Please split debug into a separate system

Reply:
Done in the latest commit.